### PR TITLE
Factor out TagMessenger and EtagGenerator ... and (deprecated) CRUD from backup

### PR DESCRIPTION
**Pull request by kaysoky**
The TagMessenger acts as a middle man that performs both etag changes and the sending of change messages.  That functionality is overloaded and should be separate.  
See: https://sagebionetworks.jira.com/browse/PLFM-2364

Since the EtagGenerator is tied closely with the TagMessenger, it is being removed along side this change
See: https://sagebionetworks.jira.com/browse/PLFM-2366

Also, many instances where these two classes are used lie within deprecated methods from V2 migration.  So those are being removed too.

### CODE REVIEW

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java [hunk 1]
```diff
@@ -119,8 +119,9 @@ private String create(Evaluation dto, Long ownerId, boolean fromBackup) {
 		copyDtoToDbo(dto, dbo);
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java:109]: remove fromBackup

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java [hunk 2]
```diff
@@ -347,7 +348,7 @@ protected static void copyDbosToDtos(List<EvaluationDBO> dbos, List<Evaluation>
 	 */
 	private void verifyEvaluationDBO(EvaluationDBO dbo) {
 		EvaluationUtils.ensureNotNull(dbo.getId(), "ID");
-		EvaluationUtils.ensureNotNull(dbo.geteTag(), "eTag");
+		EvaluationUtils.ensureNotNull(dbo.getEtag(), "etag");
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java:321]: breaking migration change without also changing how the DBO is translated to/from backup

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/SubmissionStatusDAOImpl.java [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 import java.io.IOException;
 import java.util.Date;
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/SubmissionStatusDAOImpl.java:6]: dito

**Comment by kaysoky** [comment]
Closing while adding PLFM-2366 (etag generation removal) and rerunning tests.

**Comment by kaysoky** [comment]
Removed EtagGenerator

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOActivityDAOImpl.java [hunk 4]
```diff
@@ -104,15 +107,18 @@ private String createPrivate(Activity dto, boolean forceEtag) throws DatastoreEx
 		DBOActivity dbo = new DBOActivity();
 		ActivityUtils.copyDtoToDbo(dto, dbo);
 
-		if(forceEtag){
-			if(dto.getEtag() == null) throw new IllegalArgumentException("Cannot force the use of an ETag when the ETag is null");
-			dbo.seteTag(KeyFactory.urlDecode(dto.getEtag()));
+		if (forceEtag) {
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOActivityDAOImpl.java]: The foceEtag is legacy for the v2 migration system and should be removed.

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java [hunk 5]
```diff
@@ -249,16 +234,18 @@ private String createNodePrivate(Node dto, boolean forceEtag) throws DatastoreEx
 			node.setBenefactorId(node.getId());
 		}
 
-		if(forceEtag){
-			if(dto.getETag() == null) throw new IllegalArgumentException("Cannot force the use of an ETag when the ETag is null");
+		if (forceEtag) {
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java]: again a case where "foceEtag" is legacy and no longer needed.

**Comment by kaysoky** [comment]
Removing "forceEtag" booleans

**Comment by kaysoky** [comment]
Also removing related create/update from backup (so long as they are unused (other than in tests)).

File: lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java (hunk 2 above)
Comment (kaysoky) [inline lib/jdomodels/src/main/java/org/sagebionetworks/evaluation/dao/EvaluationDAOImpl.java:321]: Note: private fields have been changed back, but the DBOs with fields of "eTag" now have both a .getEtag() and a .geteTag method (one as a default getter, the other as part of an interface used to send change messages).

**Comment by kaysoky** [comment]
Pull request got a bit messed up from rebasing after closing,

**Pull request closed without merging**